lib/public/views/QcFlags/ActiveColumns/gaqFlagsActiveColumns.js

```diff
@@ -0,0 +1,136 @@
+/**
+ * @license
+ * Copyright CERN and copyright holders of ALICE O2. This software is
+ * distributed under the terms of the GNU General Public License v3 (GPL
+ * Version 3), copied verbatim in the file "COPYING".
+ *
+ * See http://alice-o2.web.cern.ch/license for full licensing information.
+ *
+ * In applying this license CERN does not waive the privileges and immunities
+ * granted to it by virtue of its status as an Intergovernmental Organization
+ * or submit itself to any jurisdiction.
+ */
+
+import { badge } from '../../../components/common/badge.js';
+import { frontLink } from '../../../components/common/navigation/frontLink.js';
+import { tooltip } from '../../../components/common/popover/tooltip.js';
+import { qcFlagTypeColoredBadge } from '../../../components/qcFlags/qcFlagTypeColoredBadge.js';
+import { QC_SUMMARY_COLORS } from '../../../components/qcFlags/qcSummaryColors.js';
+import { formatQcFlagEnd } from '../format/formatQcFlagEnd.js';
+import { formatQcFlagStart } from '../format/formatQcFlagStart.js';
+import { h, iconBan, iconWarning } from '/js/src/index.js';
+
+/**
+ * Render display of aggregated quality for list of QC flags in a GAQ period
+ *
+ * @param {QcFlag[]} contributingFlags list of flags in given GAQ period
+ * @param {Detector[]} gaqDetectors list of all detectors which flags contribute to GAQ
+ * @return {Component} display of aggregated quality
+ */
+const formatGeneralQuality = (contributingFlags, gaqDetectors) => {
+    const flagTypes = contributingFlags.map(({ flagType }) => flagType);
+    const allGood = flagTypes.every(({ bad }) => !bad);
+    const someBadNotReproducible = flagTypes.some(({ bad, mcReproducible }) => bad && !mcReproducible);
+
+    const flagForSomeDettectorIsMissing = gaqDetectors.length !== contributingFlags.length;
+    const missingVerifications = contributingFlags.some(({ verifications }) => verifications.length === 0);
+    const missingDataMessage = [
+        flagForSomeDettectorIsMissing ? h('', 'No flag for some detector') : '',
+        missingVerifications ? h('', 'At least one flag is not verified') : '',
+    ].filter((identity) => identity);
```

Comment:
There is no need to add empty and then filter based on that. You can use the `...` operator to achieve the same

```suggestion
    const missingDataMessage = [
        ... flagForSomeDettectorIsMissing ? [h('', 'No flag for some detector')] : [],
        ... missingVerifications ? [h('', 'At least one flag is not verified')] : [],
    ];
```